docs/dev/framework/filesystem/virtual-filesystem.md

```diff
@@ -0,0 +1,292 @@
+---
+title: "Virtual filesystem"
+description: Extra metadata from the database for your file resources.
+aliases:
+  - /framework/filesystem/dbafs/
+---
+
+{{< version "4.13" >}}
+
+{{% notice warning %}}
+The new filesystem capabilities are currently considered *experimental* and therefore not covered by Contao's BC
+promise. Classes marked with `@experimental` should be considered internal for now. Although not likely, there could
+also be some behavioral changes, so be prepared.
+{{% /notice %}}
+
+
+## Names and autowiring
+Each virtual filesystem instance has a unique name which is used to generate a [named autowiring alias][NamedAliases].
+If you want to work with the `documents` filesystem, you would inject a variable named `$documentsStorage` (note the
+suffix 'Storage'):
+
+```php
+class Example {
+    public function __construct(private VirtualFilesystemInterface $documentsStorage) {}
+}
```

Comment:
Not sure if we really should use this syntax in the documentation (we don't anywhere else).

```suggestion
use Contao\CoreBundle\Filesystem\VirtualFilesystemInterface;

class Example {
    private $documentsStorage;

    public function __construct(VirtualFilesystemInterface $documentsStorage)
    {
        $this->documentsStorage = $documentsStorage;
    }
}
```

Reply:
Adjusted the other example in 17129a2ebfc934ef3c35d5c0538d9455dc4ada6f. Let's leave this one as is, because it's not a complete "copyable" example anyways.